frontend/src/scenes/trends/personsModalLogic.tsx

```diff
@@ -106,36 +108,21 @@ export const personsModalLogic = kea<personsModalLogicType<PersonModalParams>>({
             () => [preflightLogic.selectors.preflight],
             (preflight) => !!preflight?.is_clickhouse_enabled,
         ],
-    },
-    loaders: ({ actions, values }) => ({
-        people: {
-            loadPeople: async ({ peopleParams }, breakpoint) => {
-                let people = []
-                const {
-                    label,
-                    action,
-                    filters,
-                    date_from,
-                    date_to,
-                    breakdown_value,
-                    saveOriginal,
-                    searchTerm,
-                    funnelStep,
-                } = peopleParams
-                const searchTermParam = searchTerm ? `&search=${encodeURIComponent(searchTerm)}` : ''
+        peopleFilterParams: [
+            (s) => [s.peopleParams],
+            (peopleParams) => {
+                if (!peopleParams) {
+                    return null
+                }
+                const { label, action, filters, date_from, date_to, breakdown_value, funnelStep } = peopleParams
 
                 if (filters.insight === ViewType.LIFECYCLE) {
-                    const filterParams = parsePeopleParams(
+                    return parsePeopleParams(
                         { label, action, target_date: date_from, lifecycle_type: breakdown_value },
                         filters
                     )
-                    people = await api.get(`api/person/lifecycle/?${filterParams}${searchTermParam}`)
                 } else if (filters.insight === ViewType.STICKINESS) {
-                    const filterParams = parsePeopleParams(
-                        { label, action, date_from, date_to, breakdown_value },
-                        filters
-                    )
-                    people = await api.get(`api/person/stickiness/?${filterParams}${searchTermParam}`)
+                    return parsePeopleParams({ label, action, date_from, date_to, breakdown_value }, filters)
                 } else if (funnelStep || filters.funnel_viz_type === FunnelVizType.Trends) {
```

Comment:
Nit: Is it possible to split this out into two separate else if's? It seems like we're doing that in the nested if/else below and not too much code is shared.